contributors/design-proposals/pod-resource-management.md

```diff
@@ -1,114 +1,419 @@
+The `kubelet` will support manipulation of the cgroup hierarchy on
+the host using a cgroup driver. The driver is configured via the
+`--cgroup-driver` flag.
 
+The supported values are the following:
 
-There are a few shortcomings to the current model.
- - Existing QoS support does not apply to pods as a whole. On-going work to support pod level eviction using QoS requires all containers in a pod to belong to the same class. By having pod level cgroups, it is easy to track pod level usage and make eviction decisions.
- - Infrastructure overhead per pod is currently charged to the node. The overhead of setting up and managing the pod sandbox is currently accounted to the node. If the pod sandbox is a bit expensive, like in the case of hyper, having pod level accounting becomes critical.
- - For the docker runtime we have a containerd-shim which is a small library that sits in front of a runtime implementation allowing it to be reparented to init, handle reattach from the caller etc. With pod level cgroups containerd-shim can be charged to the pod instead of the machine.
- - If a container exits, all its anonymous pages (tmpfs) gets accounted to the machine (root). With pod level cgroups, that usage can also be attributed to the pod.
- - Let containers share resources - with pod level limits, a pod with a Burstable container and a BestEffort container is classified as Burstable pod. The BestEffort container is able to consume slack resources not used by the Burstable container, and still be capped by the overall pod level limits.
+* `cgroupfs` is the default driver that performs direct manipulation of the
+cgroup filesystem on the host in order to manage cgroup sandboxes.
+* `systemd` is an alternative driver that manages cgroup sandboxes using
+transient slices for resources that are supported by that init system.
 
-## Design
+Depending on the configuration of the associated container runtime,
+operators may have to choose a particular cgroup driver to ensure
+proper system behavior.  For example, if operators use the `systemd`
+cgroup driver provided by the `docker` runtime, the `kubelet` must
+be configured to use the `systemd` cgroup driver.
 
-High level requirements for the design are as follows:
- - Do not break existing users. Ideally, there should be no changes to the Kubernetes API semantics.
- - Support multiple cgroup managers - systemd, cgroupfs, etc.
+Implementation of either driver will delegate to the libcontainer library
+in opencontainers/runc.
 
-How we intend to achieve these high level goals is covered in greater detail in the Implementation Plan.
+### Conversion of cgroupfs to systemd naming conventions
 
-We use the following denotations in the sections below:
+Internally, the `kubelet` maintains both an abstract and a concrete name
+for its associated cgroup sandboxes.  The abstract name follows the traditional
+`cgroupfs` style syntax.  The concrete name is the name for how the cgroup
```

Comment:
i am not following, unless you mean the concrete name is abstract by virtue that it encodes the hierarchy at each step.  if so, i agree, that is an abstraction in and of itself.  that said, it is current reality ;-)